### PR TITLE
Bump React to version 18.

### PR DESCRIPTION
**Pull request by joebutler2**

### CODE REVIEW

**Comment by joebutler2** [comment]
This currently works as-is in my team's app, though there are warnings that do appear. 

For example:
```
bundle.js:49024 Warning: You are calling ReactDOMClient.createRoot() on a container that has already been passed to createRoot() before. Instead, call root.render() on the existing root instead if you want to update it.
    at PdfHighlighter (http://localhost:3001/static/js/bundle.js:3389:5)
    at PdfLoader (http://localhost:3001/static/js/bundle.js:3877:5)
```

This is related to rendering the highlights in [PdfHighlighter.tsx:325](https://github.com/agentcooper/react-pdf-highlighter/blob/main/src/components/PdfHighlighter.tsx#L325). Since each page has its own highlight layer we need to find a "root" that can be used to render them.

**Comment by keunbit** [comment]
> This currently works as-is in my team's app, though there are warnings that do appear.
> 
> For example:
> 
> ```
> bundle.js:49024 Warning: You are calling ReactDOMClient.createRoot() on a container that has already been passed to createRoot() before. Instead, call root.render() on the existing root instead if you want to update it.
>     at PdfHighlighter (http://localhost:3001/static/js/bundle.js:3389:5)
>     at PdfLoader (http://localhost:3001/static/js/bundle.js:3877:5)
> ```
> 
> This is related to rendering the highlights in [PdfHighlighter.tsx:325](https://github.com/agentcooper/react-pdf-highlighter/blob/main/src/components/PdfHighlighter.tsx#L325). Since each page has its own highlight layer we need to find a "root" that can be used to render them.

Thank you for your code.
but, how to avoid that warning? any update?

**Comment by joebutler2** [comment]
@keunbit that issue has been resolved with the latest version of this branch. Feel free to try it out!

File: src/components/HighlightLayer.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  Scaled,
+  ScaledPosition,
+} from "../types";
+
+interface HighlightLayerProps<T_HT> {
+  highlightsByPage: { [pageNumber: string]: Array<T_HT> };
+  pageNumber: string;
+  scrolledToHighlightId: string;
+  highlightTransform: (
+    highlight: any,
```
Comment (agentcooper) [inline src/components/HighlightLayer.tsx:17]: Could you please fix `any` types here?

**Comment by sumit-oddr** [comment]
@joebutler2 - thanks for your work. We are very interested in getting this PR merged into the release branch. Thank you!

**Comment by ayush1852017** [comment]
> @keunbit that issue has been resolved with the latest version of this branch. Feel free to try it out!

All highlights disappearing as soon as I resize PDF.
Working fine when I remove `if condition`  but getting same warning as @keunbit facing.

**Comment by pham740** [comment]
@ayush1852017 What `if` condition did you remove? I've also been running across the issue where highlights disappear upon resizing the browser window.

**Comment by pham740** [comment]
For my project, not losing the highlights when resizing the browser window is a must-have. However, when using [ayush1852017]([https://github.com/ayush1852017)'s](https://github.com/ayush1852017)%27s) fix (thank you!), the `onClick` prop of `Highlight.tsx` stops working on top of getting the same warning in the console as [keunbit](https://github.com/keunbit) and [ayush1852017](https://github.com/ayush1852017). I'm not sure whether this ties into the issue quoted below, but these would be nice features to have in this PR of an already great library.

> This is related to rendering the highlights in [PdfHighlighter.tsx:325](https://github.com/agentcooper/react-pdf-highlighter/blob/main/src/components/PdfHighlighter.tsx#L325). Since each page has its own highlight layer we need to find a "root" that can be used to render them.

**Comment by mochirfan** [comment]
I'm going to use this plugin for my new project, but in npm it's still doesn't work with react 18? because I'm getting warning about ReactDOM.render and makes text highlighting and scrolling doesn't work. Is there any chance the updated version is available in npm?

Thank you

**Comment by pham740** [comment]
@mochirfan Here is a link to the package containing this PR's code: https://github.com/orgs/credolabs/packages/npm/package/react-pdf-highlighter

You'll have to download it through the Github package registry instead of NPM, but it might be a good solution in the mean time.

**Comment by J0** [comment]
Thanks everyone for the hard work and constructive comments on the PR! Wrote a [short snippet](https://ajoellee.com/blog/2023-07-16-react-pdf-highlighter-with-nextjs-13/) about how to get up and running with the package - hope it helps in some way till the PR is merged

**Comment by siawyoung** [comment]
Thank you! Been struggling with this too. You guys are great ❤️

**Comment by k3ntar0** [comment]
Thanks @joebutler2
I encountered a problem with the installation of the npm package from forked repository, and these are the steps to resolve it.

1. Create your Personal Access Token of GitHub
2. If not exist, create  `.npmrc` and then add `//npm.pkg.github.com/:_authToken=*************`
3.  If using yarn and not exist, create `.yarnrc` and then add `"@credolabs:registry" "https://npm.pkg.github.com"`
4. After above settings, run `yarn add @credolabs/react-pdf-highlighter@5.6.7`

**Comment by pham740** [comment]
@k3ntar0 Hopefully this will give you a few leads - I'm not using yarn at all for my project so the .npmrc file I was using looked like this:

```
# Install packages from the npm registry
registry=https://registry.npmjs.org/

# Except for packages from @credolabs, which I want to install from the Github package registry
@credolabs:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken=*********************
```

From there, installing the package via the command line or package.json should work.

**Comment by agentcooper** [comment]
Thanks everyone, this is now published in version `6.0.0`.

**Comment by agentcooper** [comment]
I think there is a bug though :-) If I scroll far enough so that PDF.js removes some pages and come back to the first page, I don't see highlights.

**Pull request merged**